api/v1alpha1/clusterextension_types.go

```diff
@@ -83,6 +83,11 @@ type ClusterExtensionSpec struct {
 	//+kubebuilder:Optional
 	// Preflight defines the configuration of preflight checks.
 	Preflight *PreflightConfig `json:"preflight,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+(-[a-z0-9]+)*$
```

Comment:
We should also make this immutable as outlined in the issue and from our earlier discussions. https://github.com/operator-framework/operator-controller/pull/1011/files#diff-8e9c0eb955489f1a05a95a75a25842a010fdc02029ce06aae7d2163b895c17cbR76 should be a good example of making a field immutable